game/src/ui/hud/FullHud.cpp

```diff
@@ -409,6 +408,27 @@ FullHud::FullHud(UserInterface* ui, Framework* framework, const sf::Vector2f& si
 
     ui->getGui()->add(label);
     buttons_.emplace_back(label);
+
+
+
+    auto respawnButton = tgui::Button::create();
+    respawnButton->setRenderer(ui->getTheme()->getRenderer("MenuButton"));
+    respawnButton->setText("Respawn");
+    respawnButton->setInheritedFont(RM.getFont("default"));
+    respawnButton->setTextSize(CONF<float>("graphics/menu_button_text_size"));
+    respawnButton->setPosition(sf::Vector2f(CONF<int>("graphics/window_width_px") / 2.0f - respawnButton->getFullSize().x / 2.0f,
+                                            CONF<int>("graphics/window_height_px")) +
+                               CONF<sf::Vector2f>("graphics/respawn_pos"));
+    respawnButton->setVisible(false);
+    respawnButton->connect("mouseentered", [framework]() {
+        framework->spawnSound(RM.getSound("ui_hover"), framework->getPlayer()->getPosition()); });
+    respawnButton->connect("pressed", [framework, ui, this]() {
+        framework->spawnSound(RM.getSound("ui_upgrade"), framework->getPlayer()->getPosition(), true);
+        framework->respawn(); ui->clearWindows(); this->show(false);
+    });
+
+    ui->getGui()->add(respawnButton);
+    buttons_.emplace_back(respawnButton);
```

Comment:
Looks fine, but of course we should always avoid copy-paste. The rule is: if anything must be used two times or more in the same code it should be extracted to some other method/class. My proposal is to make class Button (maybe it should be inside common/ui/hud/ directory). It should be used to describe every button used in game (we can now assume that there are two buttons - "Back to menu" and "Respawn").  

You need to think about common parts in every button and their differences (what should be parametrized?).

Usage should look like this: 
- In FullHud there should be std::vector<Button> buttons_ (or any other name).
- In FullHud constructor you should add two buttons and initialize them. Every button needs to be connected to some function that is called after mouse pressing. My proposal is to add bindFunction method (look in the code for some examples of the same name).

Button class will help us in adding new buttons in the future.